### PR TITLE
Chore/prepare for integration

### PR DESCRIPTION
**Pull request by VanessaPC**
The Wallet API changed on Friday (night for me) so here is the new integration, directly into the SDK branch that contains that integration to get a working version, since I believe is a matter of time for it to be merged on Monday, and intending to speed things up. 
SDK  new implementation: https://github.com/input-output-hk/midnight-client-sdk/pull/27 

_If the permissions here: https://github.com/input-output-hk/midnight-client-sdk/pull/25 get fixed, then I'll push that branch. 
This branch then will point to the latest of the SDK. Which in turn will be connected to the latest of the Wallet. 
Which with all spawned system should interact with things, because I disable here all the mocked responses to actually test the system._ 

Note: I am trying to focus on getting things to work instead of the code to be nice... I am sorry about the many eye sores this cause.

### CODE REVIEW

File: pages/transfer.js [hunk 1]
```diff
@@ -9,14 +9,54 @@ const client = new Client();
...
+      //         contractId: "0x1",
+      //         payload: {
+      //           utxo: [],
+      //         },
+      //       },
+      //     ],
+      //   },
+      // };
+
+      if (awaitingTransfer && value.result.events[0].name === "transfer_sent") {
```
Comment (jmendiola222) [inline pages/transfer.js:39]: Maybe there is no need to wait for the transfer event in the "transfer" page, we can go back to the contract-display and "listen" for the transfer there.

File: pages/contract-display.js [hunk 2]
```diff
@@ -9,11 +9,50 @@ const client = new Client();
...
+      //         contractId: "0x1",
+      //         payload: {
+      //           utxo: ["Im and utxooooo"],
+      //         },
+      //       },
+      //     ],
+      //   },
+      // };
+
+      if (awaitingMint && value.result.events[0].name === "owner_minted") {
```
Comment (jmendiola222) [inline pages/contract-display.js:36]: We will also need to "react" to the "transfer_receive" event in this page. And as I suggested on the other comment, I think also for the "transfer".

File: pages/first-actions.js [hunk 3]
```diff
@@ -15,12 +15,88 @@ const joinSmartContract = async (contractId) => {
...
+        return router.push("/contract-display");
+      }
+
+      return () => {
+        subscriptionPromise.then((subscription) => subscription.unsubscribe());
+      };
+    });
+
+    return () => {
+      subscription.unsubscribe();
```
Comment (jmendiola222) [inline pages/first-actions.js:96]: I'm getting an undefined here. :thinking:

File: pages/contract-display.js (hunk 2 above)
Comment (VanessaPC) [reply pages/contract-display.js:36]: yeah sounds good

File: pages/first-actions.js (hunk 3 above)
Comment (VanessaPC) [reply pages/first-actions.js:96]: yea, the observable might have not been working. With the new changes I need a way to unsubscribe provided from the Wallet API so I have already asked for it, and I'll add it here when I get it.

**Pull request merged**